drivers/usb/device/usb_dc_nrfx.c

```diff
@@ -1402,19 +1403,21 @@ int usb_dc_ep_write(const u8_t ep, const u8_t *const data,
 	 */
 	if ((ep_ctx->cfg.type == USB_DC_EP_CONTROL)
 	    && (nrfx_usbd_last_setup_dir_get() != ep)) {
-		k_sem_give(&ctx->dma_in_use);
 		nrfx_usbd_setup_clear();
+		k_sem_give(&ctx->dma_in_use);
 		return 0;
 	}
+	ep_ctx->write_in_progress = true;
```

Comment:
maybe move it to line 1419?

can `nrfx_usbd_ep_transfer` every return an error? if it can it must be handler (not only asserted), if it can't and assert is there only for debug purposes then moving the `write_in_progress` is allowed.